### PR TITLE
torch compile config standardization update

### PR DESCRIPTION
**Pull request by agunapal**
## Description

This PR standardizes how TorchServe would be supporting any PyTorch 2.x APIs config

Define the following structure for a PyTorch 2.x feature config in `model-config.yaml`
```
pt2:
  <API name>:
    enable: True
    option1:  value1
    optionN:  valueN
```
or
```
pt2:
  <API name>:
    <feature 1>:
      enable: True
      option1:  value1
      optionN:  valueN
```

For `torch.compile`, we would specify the following options

```
pt2:
  compile:
    enable: True
    backend: "inductor"
    mode: "max-autotune"
```

This PR also fixes the issues with torch compile pytests

Fixes #(
[issue1](https://github.com/pytorch/serve/issues/3164)
[issue2](https://github.com/pytorch/serve/issues/3173)
[issue3](https://github.com/pytorch/serve/issues/3174)
)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the Unit or Integration tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

```
pytest -v test_torch_compile.py
================================================================================== test session starts ===================================================================================
platform linux -- Python 3.10.14, pytest-7.3.1, pluggy-1.0.0 -- /home/ubuntu/anaconda3/envs/torchserve/bin/python
cachedir: .pytest_cache
rootdir: /home/ubuntu/serve
plugins: cov-4.1.0, mock-3.14.0, timeout-2.3.1
collected 8 items                                                                                                                                                                        

test_torch_compile.py::TestTorchCompile::test_archive_model_artifacts PASSED                                                                                                       [ 12%]
test_torch_compile.py::TestTorchCompile::test_start_torchserve PASSED                                                                                                              [ 25%]
test_torch_compile.py::TestTorchCompile::test_server_status PASSED                                                                                                                 [ 37%]
test_torch_compile.py::TestTorchCompile::test_registered_model PASSED                                                                                                              [ 50%]
test_torch_compile.py::TestTorchCompile::test_serve_inference SKIPPED (Test failing on regression runner)                                                                          [ 62%]
test_torch_compile.py::TestTorchCompile::test_compile_inference_enable_true_default PASSED                                                                                         [ 75%]
test_torch_compile.py::TestTorchCompile::test_compile_inference_enable_true PASSED                                                                                                 [ 87%]
test_torch_compile.py::TestTorchCompile::test_compile_inference_enable_false PASSED                                                                                                [100%]

==================================================================================== warnings summary ====================================================================================
test_torch_compile.py:12
  /home/ubuntu/serve/test/pytest/test_torch_compile.py:12: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    from pkg_resources import packaging

../../../anaconda3/envs/torchserve/lib/python3.10/site-packages/pkg_resources/__init__.py:2832
../../../anaconda3/envs/torchserve/lib/python3.10/site-packages/pkg_resources/__init__.py:2832
  /home/ubuntu/anaconda3/envs/torchserve/lib/python3.10/site-packages/pkg_resources/__init__.py:2832: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('ruamel')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================================= 7 passed, 1 skipped, 3 warnings in 35.29s ========================================================================

```

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: examples/image_classifier/resnet_18/README.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ Ex:  `cd  examples/image_classifier/resnet_18`
 In this example , we use the following config
 
 ```
-echo "pt2 : {backend: inductor, mode: reduce-overhead}" > model-config.yaml
+echo "pt2:\n  compile:\n    enable: True\n    backend: inductor\n    mode: reduce-overhead" > model-config.yaml
```
Comment (mreso) [inline examples/image_classifier/resnet_18/README.md:26]: If you just use multiple lines people will be able to visualize the new format much better:
```bash
echo "pt2:
  compile:
    enable: True
    backend: inductor
    mode: reduce-overhead" > model-config.yaml
```

File: examples/pt2/torch_compile/README.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ Ex:  `cd  examples/pt2/torch_compile`
 In this example , we use the following config
 
 ```
-echo "pt2 : {backend: inductor, mode: reduce-overhead}" > model-config.yaml
+echo "pt2:\n  compile:\n    enable: True" > model-config.yaml
```
Comment (mreso) [inline examples/pt2/torch_compile/README.md:22]: See above regarding multiline echo.

File: examples/pt2/torch_compile/README.md [hunk 3]
```diff
@@ -76,7 +76,7 @@ After a few iterations of warmup, we see the following
 #### Measure inference time with `torch.compile`
 
 ```
-echo "pt2: {backend: inductor, mode: reduce-overhead}" > model-config.yaml && \
+echo "pt2:\n  compile:\n    enable: True\n    backend: inductor\n    mode: reduce-overhead" > model-config.yaml && \
```
Comment (mreso) [inline examples/pt2/torch_compile/README.md:79]: ditto

File: test/pytest/test_torch_compile.py [hunk 4]
```diff
@@ -146,3 +176,90 @@ def test_serve_inference(self):
                 "Compiled model with backend inductor, mode reduce-overhead"
                 in model_log
             )
+
+    def test_compile_inference_enable_true_default(self, chdir_example):
```
Comment (mreso) [inline test/pytest/test_torch_compile.py:180]: You should parametrize these tests to make it more obvious what is the difference between them. See https://github.com/pytorch/serve/blob/36049cb7f8046b5e72b0a790a1b0aa5b0e96aee6/test/pytest/test_example_gpt_fast.py#L98 for an example

**Review by mreso — APPROVED** [review]
LGTM, just left some nits

File: examples/pt2/torch_compile/README.md (hunk 2 above)
Comment (agunapal) [reply examples/pt2/torch_compile/README.md]: Makes sense. Done

File: examples/pt2/torch_compile/README.md (hunk 3 above)
Comment (agunapal) [reply examples/pt2/torch_compile/README.md]: Makes sense. Done

File: test/pytest/test_torch_compile.py (hunk 4 above)
Comment (agunapal) [reply test/pytest/test_torch_compile.py]: This is cool. Done

**Pull request merged**